packages/custom-functions-metadata/src/generate.ts

```diff
@@ -1,1170 +1,2 @@
-    array = getArrayDimensionalityAndTypeForArrayTypeNode(node);
-  } else if (ts.isTypeReferenceNode(node)) {
-    array = getArrayDimensionalityAndTypeForReferenceNode(node as ts.TypeReferenceNode);
-  }
-  return array;
-}
-
-/**
- * Returns the dimensionality and type of array for TypeNode
- * @param node TypeNode
- */
-function getArrayDimensionalityAndTypeForArrayTypeNode(node: ts.TypeNode): IArrayType {
-  const array: IArrayType = {
-    dimensionality: 1,
-    type: ts.SyntaxKind.AnyKeyword,
-  };
-
-  let nodeCheck = (node as ts.ArrayTypeNode).elementType;
-  array.type = nodeCheck.kind;
-  while (ts.isArrayTypeNode(nodeCheck)) {
-    array.dimensionality++;
-    nodeCheck = (nodeCheck as ts.ArrayTypeNode).elementType;
-    array.type = nodeCheck.kind;
-  }
-
-  return array;
-}
-
-/**
- * Returns the dimensionality and type of array for ReferenceNode
- * @param node TypeReferenceNode
- */
-function getArrayDimensionalityAndTypeForReferenceNode(node: ts.TypeReferenceNode): IArrayType {
-  const array: IArrayType = {
-    dimensionality: 0,
-    type: ts.SyntaxKind.AnyKeyword,
-  };
-
-  if (node.typeArguments && node.typeArguments.length === 1) {
-    let nodeCheck = node;
-    let dimensionalityCount = 1;
-    while (nodeCheck.typeArguments) {
-      // @ts-ignore
-      if (TYPE_MAPPINGS_SIMPLE[nodeCheck.typeArguments[0].kind]) {
-        array.dimensionality = dimensionalityCount;
-        array.type = nodeCheck.typeArguments[0].kind;
-      }
-      nodeCheck = nodeCheck.typeArguments[0] as ts.TypeReferenceNode;
-      dimensionalityCount++;
-    }
-  }
-
-  return array;
-}
-
-/**
- * Get the parameter dimensionality of the node
- * @param t TypeNode
- */
-function getParamDim(t: ts.TypeNode): string {
-  let dimensionality: CustomFunctionsSchemaDimensionality = "scalar";
-  if (t) {
-    if (ts.isTypeReferenceNode(t) || ts.isArrayTypeNode(t)) {
-      const array = getArrayDimensionalityAndType(t);
-      if (array.dimensionality > 1) {
-        dimensionality = "matrix";
-      }
-    }
-  }
-
-  return dimensionality;
-}
-
-function getParamOptional(p: ts.ParameterDeclaration, jsDocParamOptionalInfo: { [key: string]: string }): boolean {
-  let optional = false;
-  const name = (p.name as ts.Identifier).text;
-  const isOptional = p.questionToken != null || p.initializer != null || p.dotDotDotToken != null;
-  // If parameter is found to be optional in ts
-  if (isOptional) {
-    optional = true;
-    // Else check the comments section for [name] format
-  } else {
-    // @ts-ignore
-    optional = jsDocParamOptionalInfo[name];
-  }
-  return optional;
-}
-
-/**
- * Log containing all the errors found while parsing
- * @param error Error string to add to the log
- */
-export function logError(error: string, position?: ts.LineAndCharacter | null): string {
-  if (position) {
-    error = `${error} (${position.line + 1},${position.character + 1})`;
-  }
-  return error;
-}
+export * from "./generateCustomFunctionsMetadata";
```

Comment:
This doesn't seem like a good change. I'd like to understand what the core motivations are here.

Reply:
> allow parseTree to be used in the browser without having to polyfill fs.